File: C/temperature.c
#include <stdio.h>
#include <string.h>

int main()
{
    char scale;
    printf("Type [C] to convert for ºC or [F] to convert for ºF: ");
    scanf("%c", &scale);

    if (scale == 'C' || scale == 'c') {
        float fahrenheit;
        printf("Type the temperature in ºF: ");
        scanf("%f", &fahrenheit);

        float celsius = ((fahrenheit - 32) * 5) / 9;
        printf("%.1f ºF is %.1f ºC\n", fahrenheit, celsius);
        return(0);
    }
    else if (scale == 'F' || scale == 'f') {
        float celsius;
        printf("Type the temperature in ºC: ");
        scanf("%f", &celsius);

        float fahrenheit = (celsius * 9 / 5) + 32;
        printf("%.1f ºC is %.1f ºF\n", celsius, fahrenheit);
        return(0);
    }
    else {
        printf("Invalid temperature scale\n");
        return(0);
    }
}

File: Haskell/temperature.hs
fahrenheitToCelsius :: Float -> Float
fahrenheitToCelsius f = (f - 32) * 5 / 9

celsiusToFahrenheit :: Float -> Float
celsiusToFahrenheit c = (c * 9 / 5) + 32

main :: IO ()
main = do
  putStrLn "Choose the conversion type:"
  putStrLn "1: Fahrenheit to Celsius"
  putStrLn "2: Celsius to Fahrenheit"
  
  option <- getLine
  
  if option == "1"
    then do
      putStrLn "Enter the temperature: "
      input <- getLine
      let temperature = read input :: Float
      let result = fahrenheitToCelsius temperature
      putStrLn $ show temperature ++ "°F is equivalent to " ++ show result ++ "°C"
  else if option == "2"
    then do
      putStrLn "Enter the temperature: "
      input <- getLine
      let temperature = read input :: Float
      let result = celsiusToFahrenheit temperature
      putStrLn $ show temperature ++ "°C is equivalent to " ++ show result ++ "°F"
  else
    putStrLn "Invalid option. Please run the program again and choose 1 or 2."